src/components/AddItem.jsx

```diff
@@ -25,16 +26,25 @@ function AddItem() {
   const [frequency, setFrequency] = useState(7);
   const [notification, setNotification] = useState('');
 
+  const navigate = useNavigate();
+
   //retrive the token from localStorage
-  const token = localStorage.getItem('list-token');
+  const localToken = localStorage.getItem('list-token');
+
+  useEffect(() => {
+    if (!localToken) {
+      navigate('/');
+      return;
+    }
+  });
 
   //handle submit for the form
   const handleSubmit = async (e) => {
     e.preventDefault();
     //create new doc reference for text input and save to firestore database
     //use a token to save new items into specific collection (shopping list)
     try {
-      const docRef = await addDoc(collection(db, token), {
+      const docRef = await addDoc(collection(db, localToken), {
```

Comment:
❤️ the specificity in variable names (token -> localToken), and perfect for this function. It seems obvious at the time what the token is about, but for future you or reviewers, it's super useful. If this was used across project, you could go as crazy as databaseLocalStorageToken or such.